libs/ml/include/ml/core/graph.hpp

```diff
@@ -113,9 +115,10 @@ class Graph
   void LinkNodesInGraph(std::string const &node_name, std::vector<std::string> const &inputs);
 
 protected:
-  std::unordered_map<std::string, NodePtrType> nodes_;
-  std::unordered_map<std::string, SizeType>    trainable_lookup_;
-  std::vector<NodePtrType>                     trainable_nodes_;
+  void InsertSharedCopy(std::shared_ptr<Graph<TensorType>> output_ptr);
```

Comment:
ultimate level of pedantry but I would usually put variables first then function declarations second (under each scope: public/protected/private)